src/utils/install.jl

```diff
@@ -0,0 +1,58 @@
+# Install a GAP package via the URL of a GIT repository.
+# If the package was already installed then update the installation,
+# otherwise clone the repository.
+# (The code is a modified copy of GAP's `InstallPackageFromGit`.)
+function GAP_Packages_install(url::String; interactive::Bool = false, branch::String = "")
+  # point PackageManager to GAP.jl's pkg dir
+  GAP.Globals.PKGMAN_CustomPackageDir = GapObj(GAP.Packages.DEFAULT_PKGDIR[])
+  # avoid info messages from PackageManager
+  GAP.Globals.SetInfoLevel(GAP.Globals.InfoPackageManager, 0)
+
+  url = GapObj(url)
+  name = GAP.Globals.PKGMAN_NameOfGitRepo(url)
+  name === GAP.Globals.fail && return false
+  dir = GAP.Globals.Filename(GAP.Globals.Directory(GAP.Globals.PKGMAN_PackageDir()), name)
+
+  # check for existing repository
+  allinfo = GAP.Globals.PackageInfo(name)
+  info = GAP.Globals.Filtered(allinfo,
+             x -> GAP.Globals.StartsWith(x.InstallationPath, GAP.Globals.PKGMAN_PackageDir()))
+  dirs = GAP.Globals.List(info, i -> i.InstallationPath)
+  repo = GAP.Globals.Filename(GAP.Globals.List(dirs, GAP.Globals.Directory), GapObj(".git"));
+  if repo !== GAP.Globals.fail
+    # the package is already installed, update it
+    return GAP.Globals.UpdatePackage(name, interactive)
+  end
+
+  ! GAP.Globals.PKGMAN_IsValidTargetDir(dir) && return false
+  if branch == ""
+    exec = GAP.Globals.PKGMAN_Exec(GapObj("."), GapObj("git"), GapObj("clone"), url, dir)
```

Comment:
This code is indeed a very 1-1 translation of the GAP code, otherwise bits like this here really should have been written as
```suggestion
    exec = run(`git clone $url $dir`)
```
(after changing `url` and `dir` to be Julia strings, of course).

But given that this function is meant as a temporary hack, since you already wrote and tested it this way, let's keep it that way then

Reply:
My idea was that we need something a bit different from what PackageManager provides, and that there will be a few necessary improvements which then might be carried back to PackageManager. Let us see.

Reply:
of course